packages/mountable/src/dune/utils/fileUtils.spec.ts

```diff
@@ -95,4 +99,113 @@ describe("fileUtils", () => {
+    });
+
+    it("should return false for non-zip files", () => {
+      expect(isZipFile("test.txt")).toBe(false);
+      expect(isZipFile("test.py")).toBe(false);
+      expect(isZipFile("test", "text/plain")).toBe(false);
+      expect(isZipFile("test", "application/json")).toBe(false);
+    });
+
+    it("should return false when no MIME type provided and no .zip extension", () => {
+      expect(isZipFile("test")).toBe(false);
+      expect(isZipFile("test.tar.gz")).toBe(false);
+    });
+  });
+
+  describe("processZipFile", () => {
+    const mockBinaryData = new ArrayBuffer(1024);
+    const mockFileName = "test.zip";
+
+    // Mock Entry type for testing
+    const createMockFileEntry = (
+      filename: string,
+      content: string,
+    ): ZipEntry => {
+      const encodedContent = new TextEncoder().encode(content);
+      return {
+        filename,
+        directory: false,
+        getData: jest.fn().mockResolvedValue(encodedContent),
+      };
+    };
+
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it("should process zip file with multiple text files successfully", async () => {
+      const mockFile1 = createMockFileEntry("file1.txt", "Hello from file1");
+      const mockFile2 = createMockFileEntry(
+        "file2.py",
+        "print('Hello from file2')",
+      );
+      const mockFile3 = createMockFileEntry(
+        "subdir/file3.js",
+        "console.log('Hello from file3')",
+      );
+
+      const mockEntries = [mockFile1, mockFile2, mockFile3];
+
+      const mockGetZipEntries: typeof defaultGetZipEntries = jest
+        .fn()
+        .mockResolvedValue(mockEntries);
+      const result = await processZipFile(
+        mockBinaryData,
+        mockFileName,
+        mockGetZipEntries,
+      );
+
+      expect(mockGetZipEntries).toHaveBeenCalledWith(mockBinaryData);
+      expect(result).toEqual({
+        "file1.txt": "Hello from file1",
+        "file2.py": "print('Hello from file2')",
+        "subdir/file3.js": "console.log('Hello from file3')",
+      });
+
+      // Verify that getData was called for files but not directories
+      expect(mockFile1.getData).toHaveBeenCalled();
+      expect(mockFile2.getData).toHaveBeenCalled();
+      expect(mockFile3.getData).toHaveBeenCalled();
+    });
+
+    it("should throw error when file extraction fails", async () => {
+      const mockFile1 = createMockFileEntry("file1.txt", "Success content");
+      const mockFile3 = createMockFileEntry("file3.txt", "Another success");
+
+      // Create a file entry that will fail
+      const mockFile2: ZipEntry = {
+        filename: "file2.txt",
+        directory: false,
+        getData: jest.fn().mockRejectedValue(new Error("Extraction failed")),
+      };
+
+      const mockEntries = [mockFile1, mockFile2, mockFile3];
+
+      const mockGetZipEntries: typeof defaultGetZipEntries = jest
+        .fn()
+        .mockResolvedValue(mockEntries);
```

Comment:
You don't really need a full mock here
```suggestion
        () => Promise.resolve(mockEntries);

```